### PR TITLE
364 process system

### PR DESCRIPTION
**Pull request by Moopli**
Incomplete process system code, but also has useful refactoring to compound and process registries

### CODE REVIEW

**Comment by TheCreator--** [comment]
Okay, so I have good news and bad news. The good news is that the game didn't crash no matter what I did :p

Unfortunately, I don't think that your process system is working yet. After playing for around 30 seconds, my cell dies because it has no ATP, even though I had around 30 oxygen and 30 glucose. After adding a couple extra vacuoles and mitochondria and collecting close to a hundred of oxygen and glucose, my cell began producing ATP as expected, but even after 5 minutes of playthrough, I had 0 reproductase. I also noticed that my cell had CO2 (though not as much as in previous versions). Shouldn't all CO2 be ejected since we don't have any processes that use it?

**Comment by Moopli** [comment]
Todo note: refactor away self.microbe.compounds, figure out a clean way to iterate through CompoundBagComponent instead. Probably good to move those loops to a System.

**Comment by jjonj** [comment]
Let me know when you want me to take a proper look!

**Comment by Moopli** [comment]
@jjonj now's a good time

**Comment by jjonj** [comment]
Taking a look!

File: scripts/microbe_stage/configs.lua [hunk 1]
```diff
@@ -55,12 +55,18 @@ compounds = {
...
     oxytoxy = {
         name = "OxyToxy NT",
+        weight = 1,
         mesh = "oxytoxy.mesh",
         size = 0.3,
-        effect = "oxytoxyEffect",
+        effect = oxytoxyEffect,
         --[[
         we'll have to be careful with this referencing. 
         should we link the function directly? Or just name?
```
Comment (jjonj) [inline scripts/microbe_stage/configs.lua]: Comment no longer relevant!

File: scripts/microbe_stage/configs.lua [hunk 2]
```diff
@@ -131,6 +133,43 @@ processes = {
     },
 }
 
+-- currently only stores process capacity data for processing organelles, but can later also store material costs for organelles
+organelles = {
+    nucleus = {
+        processes = {
+            ReproductaseSynthesis = 1,
```
Comment (jjonj) [inline scripts/microbe_stage/configs.lua]: Is the "1" here just to mean true?

File: scripts/microbe_stage/microbe.lua [hunk 3]
```diff
@@ -37,11 +37,11 @@ function MicrobeComponent:__init(isPlayerMicrobe, speciesName)
...
-    self.compoundPriorities = {}
-    self.defaultCompoundPriorities = {}
-    self.defaultCompoundPriorities[CompoundRegistry.getCompoundId("atp")] = 10
-    self.defaultCompoundPriorities[CompoundRegistry.getCompoundId("reproductase")] = 8
-    self:_resetCompoundPriorities()
+    -- self.compoundPriorities = {}
+    -- self.defaultCompoundPriorities = {}
+    -- self.defaultCompoundPriorities[CompoundRegistry.getCompoundId("atp")] = 10
+    -- self.defaultCompoundPriorities[CompoundRegistry.getCompoundId("reproductase")] = 8
+    -- self:_resetCompoundPriorities()
```
Comment (jjonj) [inline scripts/microbe_stage/microbe.lua]: Still need these here commented out?

File: scripts/microbe_stage/microbe.lua [hunk 4]
```diff
@@ -313,6 +324,10 @@ function Microbe:__init(entity)
     end
     if not self.microbe.initialized then
         self:_initialize()
+        if in_editor == nil then
```
Comment (jjonj) [inline scripts/microbe_stage/microbe.lua]: == nil or == false

File: scripts/microbe_stage/configs.lua (hunk 2 above)
Comment (Moopli) [inline scripts/microbe_stage/configs.lua]: No, this is a measure of how much capacity the organelle has to carry out the process

File: scripts/microbe_stage/configs.lua (hunk 1 above)
Comment (Moopli) [inline scripts/microbe_stage/configs.lua]: As are a lot of other comments, and a lot of commented-out or dead code :P
I'll be fixing it all once it looks like the branch is ready.

File: scripts/microbe_stage/microbe.lua (hunk 4 above)
Comment (Moopli) [inline scripts/microbe_stage/microbe.lua]: I remember having a very annoying bug, probably due to missing something obvious, which I fixed by making this check if the parameter exists. I can revisit it though.

**Pull request merged**